### PR TITLE
feat: Add support for windows

### PR DESCRIPTION
**Pull request by jsirianni**
Added support for Windows. Tested to ensure Linux support does not contain regressions.

Tested against BindPlane Cloud with agent version 1.28 and then 1.30 (upgrade in place using ansible). During upgrade, Ansible correctly restarts the service.
```
- name: bindplane-agent
  hosts: all
  #remote_user: joe_sirianni
  #become: yes
  roles:
    - role: bindplane_agent
      version: "1.30.0"
      endpoint: "wss://app.bindplane.com/v1/opamp"
      secret_key: redacted
      labels: configuration=windows
```

Ansible output:
```
PLAY [bindplane-agent] 

TASK [Gathering Facts] 
ok: [34.68.72.42]

TASK [bindplane_agent : Validate version] 
ok: [34.68.72.42] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [bindplane_agent : Validate endpoint] 
ok: [34.68.72.42] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [bindplane_agent : Validate secret_key] 
ok: [34.68.72.42] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [bindplane_agent : Call Linux specific playbook] 
skipping: [34.68.72.42]

TASK [bindplane_agent : Call Windows specific playbook] 
included: /home/jsirianni/git/bindplane-agent-ansible/tasks/windows.yml for 34.68.72.42

TASK [bindplane_agent : Install MSI package] 
changed: [34.68.72.42]

TASK [bindplane_agent : Check if manager configuration exists] 
ok: [34.68.72.42]

TASK [bindplane_agent : Create initial manager configuration] 
changed: [34.68.72.42]

RUNNING HANDLER [bindplane_agent : restart windows service] 
changed: [34.68.72.42]

PLAY RECAP 
34.68.72.42                : ok=9    changed=3    unreachable=0    failed=0    skipped=1    rescued=0    ignored=0   

```

BindPlane shows the agent with the correct version and labels.

Screenshot from 2023-07-28 10-42-55

Repeat Ansible runs are idempotent.
```
34.68.72.42                : ok=7    changed=0    unreachable=0    failed=0    skipped=2    rescued=0    ignored=0   
```

### CODE REVIEW

File: defaults/main.yml [hunk 1]
```diff
@@ -8,4 +8,5 @@ insecure_skip_verify: false
 
 package_type: "{{ 'rpm' if ansible_os_family in ['RedHat', 'Suse'] else 'deb' }}"
 arch: "{{ 'arm64' if ansible_architecture in ['arm64', 'aarch64'] else 'amd64' }}"
-package_url: "https://github.com/observIQ/observiq-otel-collector/releases/download/v{{ version }}/observiq-otel-collector_v{{ version }}_linux_{{ arch }}.{{ package_type }}"
+linux_package_url: "https://github.com/observIQ/observiq-otel-collector/releases/download/v{{ version }}/observiq-otel-collector_v{{ version }}_linux_{{ arch }}.{{ package_type }}"
+windows_package_url: "https://github.com/observIQ/observiq-otel-collector/releases/download/v{{ version }}/observiq-otel-collector.msi"
```
Comment (cpheps) [inline defaults/main.yml]: Probably should use the bindplane-agent urls

**Review by cpheps — APPROVED** [review]

File: tasks/windows.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+---
+- name: Install MSI package
+  ansible.windows.win_package:
+    path: "{{ windows_package_url }}"
+    product_id: D67CCA1A-6708-4096-8BDE-5069739FB861
```
Comment (BinaryFissionGames) [inline tasks/windows.yml]: This product ID is different for every release I believe, I think we can just remove this? 
It says it's ignored when path is a not exe and is a local accesible file path.

https://docs.ansible.com/ansible/latest/collections/ansible/windows/win_package_module.html#parameter-product_id

Comment (BinaryFissionGames) [reply tasks/windows.yml]: Actually, this is a URL, not sure what the implications of having this set different from the actual package's product ID is.

Comment (jsirianni) [reply tasks/windows.yml]: Gool call. I gave it a spin and it works great without product_id.

**Review by BinaryFissionGames — APPROVED** [review]

**Pull request merged**